api/base/serializers.py

```diff
@@ -171,7 +174,16 @@ def get_attribute(self, obj):
         return obj
 
     def to_representation(self, obj):
-        ret = _rapply(self.links, _url_val, obj=obj, serializer=self.parent)
+        ret = {}
+        for name, value in self.links.iteritems():
+            try:
```

Comment:
Instead of having the try-except block raise (and handle) its own exception, I recommend consolidating this- in addition to being shorter and more readable, it avoids a situation where multiple statements in a try block can make code harder to debug.

```python
url = _url_val(value, obj=obj, serializer=self.parent)
if not url:
    continue
else:
    ret[name] = url
```